Release/v0.1.0

ggsdc opened the pull request.

ggsdc committed:
Updated requirements and workflows (#48)

* Updated requirements and workflows

* Fixed workflow file

* Run if workflow changes

* Cache pip dependencies

* More changes to cache

* Changed action version

* Small typo

* Fix typo on workflow

hugolarzabal committed:
fix left_join and add test (#46)

* fix left_join and add test

* add if_empty_table argument to left join

* add if_empty_table argument to joins

* complete docstrings

---------

Co-authored-by: Guillermo González-Santander de la Cruz <guillermo.gonzalez@baobabsoluciones.es>

hugolarzabal committed:
change load_excel function to accept dtype and output as list of dict (#49)

hugolarzabal committed:
Feature/new date functions (#47)

* new date functions (still work in progress)

* new date functions and tests

---------

Co-authored-by: Guillermo González-Santander de la Cruz <guillermo.gonzalez@baobabsoluciones.es>

ggsdc committed:
Feature/drop images (#50)

* Drop images module.
Rearranged dependencies.
Test build on github action

* Extra dependencies also come from requirements file

* Added TODO comments

* Expand tests to macOS

hugolarzabal committed:
Feature/remove pandas (#52)

* try to refactor read and write excel without pandas

* prepare function to write excel with openpyxl

* create is_null function and stop using pandas function

* remove pandas in tests

* remove pandas dependency from file_functions (some function need alternatives without pandas)

* apply black

* optional import for mango processing

* remove unused pandas dependencies

* add csv function which do not use pandas

* fix test for null values

---------

Co-authored-by: Guillermo Gonzalez-Santander <guillermo.gonzalez@baobabsoluciones.es>

ggsdc committed:
Changed from setup.py to pyproject.toml (#54)

* Changed from setup.py to pyproject.toml

* Adapted worflow for testing + some small changes

* Added pydantic validation akin to current fastjsonschema valdiation (#51)

* Drop images module.
Rearranged dependencies.
Test build on github action

* Extra dependencies also come from requirements file

* Added TODO comments

* Expand tests to macOS

* Added pydantic validation akin to current fastjsonschema valdiation

* Some import statements corrected

* Updated pyproject.toml

* refactor workflows

* Change names

* Modified table coverage source

ggsdc committed:
Bump version

ggsdc committed:
Fixed workflow

ggsdc committed:
Fixed workflow

ggsdc committed:
Deleted unused filter

ggsdc committed:
Stop using cache

AntonioGonzalezSuarez commented on mango/shared/decorators.py:
```diff
@@ -39,3 +41,33 @@ def wrapper(*args, **kwargs):
     return decorator
+
+
+def pydantic_validation(**validators):
```
Example of usage? I dont get how i can use it. In my mind you would do something like this but doesnt seem to be what it does:

```python
from pydantic import BaseModel
from mango.shared import pydantic_validation


class Dummy(BaseModel):
	a: int
	b: str
	c: float
	

@pydantic_validation(Dummy)
def dummy_function(a:int,b:str,c:float):
	pass
	
# Should raise errors
dummy_function(a="1",b=1,c=1)
```

However, as it works now the arguments of dummy function must be a dictionary and then one validator for each argument:

```python
@pydantic_validation(first_arg=Dummy)
def dummy_function(first_arg: dict):
	pass
	
# No error
dummy_function(first_arg={"a":1,"b":"1","c":1.0})
# Error (a is a string)
dummy_function(first_arg={"a":"1","b":"1","c":1.0})
```

I think can be abstracted all cases by changing how the validators are handled

ggsdc replied:
In my initial approach I thought that having only valdiators for the lists or dicts that where going to be passed as arguments would be enough, that validating a simple string or float would be too much, but it can be changed and adapted to work as you say.

AntonioGonzalezSuarez replied:
OK for this release we can keep it like this I will add it as an issue on repo

hugolarzabal commented on mango/processing/__init__.py:
```diff
@@ -5,12 +5,16 @@
     is_json_file,
```
We are missing a statement:
`from .date_functions_2 import *`

ggsdc replied:
IS there any reason to have the date functions split between two files instead of just one file?

hugolarzabal replied:
no, they could go in the same file.

gvalleg committed:
Feature/add calendars datasets (#56)

* Get calendar with holidays library.

* Added workflow for data tests

* Wroing import statement

* Update requirements

* Change date to datetime

* Fix assert

* Improve Codecov

* Fix assertRaises

---------

Co-authored-by: Guillermo Gonzalez-Santander <guillermo.gonzalez@baobabsoluciones.es>

ggsdc committed:
Missing import

ggsdc committed:
Refactor date functions

ggsdc committed:
Merge branch 'development' into release/v0.1.0

# Conflicts:
#	pyproject.toml

ggsdc committed:
Merge development and changed requirements file

antoniogonzalezsuarez committed:
Feature/meteorological data (#58)

* Implemented Aemet class in aemet.py. It will allow to get meteorological
for a given location or for all stations in Spain

* Added sleep to not saturate API

* API Key invalid exceptions.py
Added more sleep to avoid overloading AEMET API. (Limit is 50 per minute)

* requirements.txt updated

* Base Class for RestClient
Added Aemet forecast endpoint

* Blacked and encoding

* Change signature of abstract method to be more generic
Corrected typo

* Refactor validators

* Finished testing of new client

* Blacked and optimized imports

* Updated requirements.txt
Added typehint where missing
Improved docstrings

* Fix type annotation error for python 3.8

* Fix type annotation error for python 3.8

* Fix type annotation error for python 3.8

* Changed class names.
Removed load_dotenv
Moved setting of defaults into __init__

* Changed name of method in rest_client.py
Added expected status "decorator" from cornflow

* Added tqdm to pyproject.toml

---------

Co-authored-by: Guillermo Gonzalez-Santander <guillermo.gonzalez@baobabsoluciones.es>

hugolarzabal committed:
Feature/improve tables (#59)

* various improvement to Table class

* add to_excel method in table

* improve tests

---------

Co-authored-by: Guillermo González-Santander de la Cruz <guillermo.gonzalez@baobabsoluciones.es>

gvalleg committed:
Feature/improve calendar data (#61)

* Remove national holiday jueves santo

* Generate dict to assign weights

* Add weight column

* Fix test

* Small fix on workflow

* Another small fix on workflows

---------

Co-authored-by: Guillermo González-Santander de la Cruz <guillermo.gonzalez@baobabsoluciones.es>

antoniogonzalezsuarez committed:
Feature/60 generalize pydantic validation for primitive type (#62)

* Pydantic decorator updated to be used in both ways

* Documented decorators.py

---------

Co-authored-by: Guillermo González-Santander de la Cruz <guillermo.gonzalez@baobabsoluciones.es>

ggsdc committed:
Merge branch 'development' into release/v0.1.0

# Conflicts:
#	pyproject.toml
#	requirements.txt

ggsdc committed:
Updated changelog

antoniogonzalezsuarez committed:
Added more params to plotly function (#64)

* Added more params to plotly function

* Support for encoding in join_html

* Added context manager to handle files

* Check the end of the output path to save.

---------

Co-authored-by: Guillermo Valle <guillermo.valle@baobabsoluciones.es>

hugolarzabal committed:
Bugfix/fix release 0.10 (#65)

* fix tests for load and write excel light. fix to_tz function

* fix test for write_excel

---------

Co-authored-by: Guillermo González-Santander de la Cruz <guillermo.gonzalez@baobabsoluciones.es>

gvalleg committed:
Feature/file explorer app (#66)

* File selector and render the file

* Mango Explorer App - v1

* Mango Explorer App - changes

* Add line between rows

* first cli approach

* Class FileExplorerApp OK

* cli second approach

* cli second approach

* cli second approach

* Try to change command path

* Wrong file name

* Add logo.png

* Add checks

* Fix empty layout dict in when argument editable == 0

* Fix empty config_path

* Disable paths

* Fix .html rendering

* Add docstring

* Add height and width plot config

* Change to .get() in dicts

* Fix length problem

---------

Co-authored-by: Guillermo Gonzalez-Santander <guillermo.gonzalez@baobabsoluciones.es>

ggsdc committed:
Merge branch 'development' into release/v0.1.0

# Conflicts:
#	mango/processing/date_functions_2.py
#	mango/tests/processing_module/test_date_functions_2.py
#	requirements.txt

ggsdc committed:
Merge changes

gvalleg committed:
Feature/file explorer app (#67)

* File selector and render the file

* Mango Explorer App - v1

* Mango Explorer App - changes

* Add line between rows

* first cli approach

* Class FileExplorerApp OK

* cli second approach

* cli second approach

* cli second approach

* Try to change command path

* Wrong file name

* Add logo.png

* Add checks

* Fix empty layout dict in when argument editable == 0

* Fix empty config_path

* Disable paths

* Fix .html rendering

* Add docstring

* Add height and width plot config

* Change to .get() in dicts

* Fix length problem

* Remove print

---------

Co-authored-by: Guillermo Gonzalez-Santander <guillermo.gonzalez@baobabsoluciones.es>

ggsdc committed:
Merge branch 'development' into release/v0.1.0

ggsdc committed:
updated changelog

ggsdc committed:
Updated pyproject.toml

gvalleg committed:
Fix error and implement checkbox not editable

gvalleg committed:
Black

gvalleg committed:
Add custom columns for row

gvalleg committed:
Fix checkbox editable

gvalleg committed:
Change logic editable checkbox

gvalleg committed:
Fix save configuration

gvalleg committed:
Changes logic

gvalleg committed:
Fix duplicates plots

antoniogonzalezsuarez committed:
Merge pull request #69 from baobabsoluciones/feature/file_explorer_app

Feature/file explorer app

antoniogonzalezsuarez committed:
Merge branch 'development' into release/v0.1.0
